SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Storage/StorageAdapter.cs

```diff
@@ -151,6 +153,92 @@ private List<object[]> ExecuteSelectionSqlQuery(string query, IList<object> args
             }
         }
 
+        private long GetMaxPageCount()
+        {
+            return GetPragmaValue("max_page_count");
+        }
+
+        private long GetPageCount()
+        {
+            return GetPragmaValue("page_count");
+        }
+
+        private long GetPageSize()
+        {
+            return GetPragmaValue("page_size");
+        }
+
+        private long GetPragmaValue(string valueName)
+        {
+            var result = ExecuteSelectionSqlQuery($"PRAGMA {valueName};");
+            var count = (long)(result.FirstOrDefault()?.FirstOrDefault() ?? 0L);
+            return count;
+        }
+
+        public long GetMaxStorageSize()
+        {
+            try
+            {
+                return GetMaxPageCount() * GetPageSize();
+            }
+            catch (StorageException)
+            {
+                AppCenterLog.Error(AppCenterLog.LogTag, $"Could not get max storage size.");
+                return -1;
+            }
+        }
+
+        public bool SetMaxStorageSize(long sizeInBytes)
+        {
+            bool success;
+            var db = _db ?? throw new StorageException("The database wasn't initialized.");
+
+            // Check the current number of pages in the database to determine whether the requested size will shrink the database.
+            var currentPageCount = GetPageCount();
+            var pageSize = GetPageSize();
+            AppCenterLog.Info(AppCenterLog.LogTag, $"Found {currentPageCount} pages in the database.");
+            var requestedMaxPageCount = Convert.ToBoolean(sizeInBytes % pageSize) ? sizeInBytes / pageSize + 1 : sizeInBytes / pageSize;
+            if (currentPageCount > requestedMaxPageCount)
+            {
+                AppCenterLog.Warn(AppCenterLog.LogTag, $"Cannot change database size to {sizeInBytes} bytes as it would cause a loss of data. " +
+                    "Maximum database size will not be changed.");
+                success = false;
```

Comment:
I guess it's already `false` (by default)

Reply:
Why do we need this flag? we can just `return false` here and in similar places because everywhere it's the last operation before return